earth_enterprise/src/common/FileAccessor.h

```diff
@@ -36,26 +39,12 @@ class AbstractFileAccessor {
   virtual void fprintf(const char *format, ...) = 0;
 };
 
-class POSIXFileAccessor: public AbstractFileAccessor {
-  int fileDescriptor;
-public:
-  POSIXFileAccessor(int i = -1) : fileDescriptor{ i }{}
-  ~POSIXFileAccessor(void) {
-    if (this->isValid()) {
-      this->Close();
-    }
-  }
+class FileAccessorFactory {
+  friend class FileAccessorPluginLoader;
```

Comment:
Will every class that inherits from this one need to make the same `friend` declaration? If so, that's a bit unintuitive. However, since the `FileAccessorPluginLoader` only deals directly with this base class, maybe only this class needs the `friend` declaration. I guess it's not that big of a deal either way.

Reply:
I don't believe they have to since the code is working with base class pointers. At least it didn't seem to in my testing.